### PR TITLE
New example: toggle between interface and implementation

### PR DESCRIPTION
**Pull request by svdo**
Please answer the following questions and leave the below in as part of your PR.

- [X] I have read the [developer documentation](https://github.com/BetterThanTomorrow/joyride/blob/master/CONTRIBUTE.md).

**I didn't do the following, since this is just a new example:**

- [ ] This PR corresponds to an [issue with a clear problem statement](https://github.com/babashka/babashka/blob/master/doc/dev.md#start-with-an-issue-before-writing-code).

- [ ] This PR contains a [test](https://github.com/BetterThanTomorrow/joyride/tree/master/vscode-test-runner/workspace-1/.joyride/src/integration_test) to prevent against future regressions

- [ ] I have updated the **\[Unreleased\]** section of the [CHANGELOG.md](https://github.com/BetterThanTomorrow/joyride/blob/master/CHANGELOG.md) file with a link to the addressed issue.

### CODE REVIEW

File: examples/.joyride/scripts/toggle_between_interface_and_impl.cljs
Comment (PEZ) [inline examples/.joyride/scripts/toggle_between_interface_and_impl.cljs:66]:
```suggestion
(when (= (joyride/invoked-script) joyride/*file*)
  (main))
```

Unless there's a reason for the current setup?

Comment (PEZ) [reply examples/.joyride/scripts/toggle_between_interface_and_impl.cljs:66]: This will let you load the file in the REPL without it being run, but will always be run when invoked as a script.

**Comment by PEZ** [comment]
Thanks! 🙏 

I only had a small suggestion about how to make the script REPL-loadable without jazzing with state.

**Comment by svdo** [comment]
Perfect Peter, no I didn't have a reason for that, just some very old example code that got copied over.

**Comment by svdo** [comment]
Is there some `require` that I can add to fix the linter warnings about the joyride ns?

**Comment by svdo** [comment]
Hmm, actually, now I get:

`ERROR: Run Failed: toggle_between_interface_and_impl.cljs Could not resolve symbol: joyride/invoked-script`

File: examples/.joyride/scripts/toggle_between_interface_and_impl.cljs
Comment (PEZ) [inline examples/.joyride/scripts/toggle_between_interface_and_impl.cljs:20]:
```suggestion
            [clojure.string :as str]
            [joyride.core :as joyride]
```

This is why you get that error. 😀

**Comment by PEZ** [comment]
> Is there some `require` that I can add to fix the linter warnings about the joyride ns?

You need to require joryride.core. Not just to please the linter. See my comment in the code.

File: examples/.joyride/scripts/toggle_between_interface_and_impl.cljs
Comment (PEZ) [inline examples/.joyride/scripts/toggle_between_interface_and_impl.cljs:32]:
```suggestion
```

Not needed any longer, iiuc.

**Comment by svdo** [comment]
Ok that was also a left-over from the original starting point that I took.

By the way I converted this PR to draft, because I've been adding support for more cases in my dev env, so I'll update it soon.

**Comment by PEZ** [comment]
Ah, I wondered about the draft thing when I got notified. 😄 That's awesome. I'm here if you need a second pair of eyes on anything, or wonder about something.

**Comment by svdo** [comment]
I added another feature; it's not "complete" but good enough for me at the moment, and I don't expect to have time to further improve it in the short run. I'll finish the PR for now so that others may also benefit from it. If I make more improvements in the future, I'll create a new PR to this repo as well.

**Comment by PEZ** [comment]
Sounds great. Not sure you meant that this PR is good to go now? Let me know when you think it is and I'll have a look and probably merge.

**Comment by svdo** [comment]
Yeah please do. It's as ready as it's going to be for now, so as far as I'm concerned this can be reviewed and merged. Thanks!

**Review by PEZ — APPROVED** [review]
Awesome. Thanks!!! 🙏 ❤️

**Pull request merged**
